x-pack/filebeat/input/netflow/integration_test.go

```diff
@@ -0,0 +1,388 @@
+							"detect_sequence_reset": true,
+							"max_message_size":      "10KiB",
+						}),
+					},
+				},
+			},
+		},
+	}
+
+	healthyChan := make(chan struct{})
+	server := &mock.StubServerV2{
+		CheckinV2Impl: func(observed *proto.CheckinObserved) *proto.CheckinExpected {
+
+			if healthyChan != nil {
+				unitState, payload := extractStateAndPayload(observed, "input-unit-1")
+				if unitState == proto.State_HEALTHY {
+					if payload.streamStatusEquals("netflow-netflow.netflow-1e8b33de-d54a-45cd-90da-23ed71c482e2", map[string]interface{}{
+						"status": "HEALTHY",
+						"error":  "",
+					}) {
+						close(healthyChan)
+						healthyChan = nil
+					}
+				}
+			}
+
+			return &proto.CheckinExpected{
+				Units: allStreams,
+			}
+		},
+		ActionImpl: func(response *proto.ActionResponse) error {
+			return nil
+		},
+	}
+	if err := server.Start(); err != nil {
+		t.Fatalf("failed to start StubServerV2 server: %v", err)
+	}
+	defer server.Stop()
+
+	// It's necessary to change os.Args so filebeat.Filebeat() can read the
+	// appropriate args at beat.Execute().
+	initialOSArgs := os.Args
+	os.Args = []string{
+		"filebeat",
+		"-E", fmt.Sprintf(`management.insecure_grpc_url_for_testing="localhost:%d"`, server.Port),
+		"-E", "management.enabled=true",
+		"-E", "management.restart_on_output_change=true",
+		"-E", "logging.level=info",
+	}
+	defer func() {
+		os.Args = initialOSArgs
+	}()
+
+	beatCmd := filebeat.Filebeat()
+	beatRunErr := make(chan error)
+	go func() {
+		defer close(beatRunErr)
+		beatRunErr <- beatCmd.Execute()
+	}()
+
+	select {
+	case <-healthyChan:
+		break
+	case err := <-beatRunErr:
+		t.Fatalf("beat run err: %v", err)
+	case <-time.After(10 * time.Second):
+		t.Fatalf("timed out waiting for beat to become healthy")
+	}
+
+	registry := monitoring.GetNamespace("dataset").GetRegistry().GetRegistry("netflow_integration_test")
+
+	discardedEventsTotalVar, ok := registry.Get("discarded_events_total").(*monitoring.Uint)
+	require.True(t, ok)
+
+	receivedEventTotalVar, ok := registry.Get("received_events_total").(*monitoring.Uint)
+	require.True(t, ok)
+
+	udpAddr, err := net.ResolveUDPAddr("udp", "127.0.0.1:6006")
+	require.NoError(t, err)
+
+	conn, err := net.DialUDP("udp", nil, udpAddr)
+	require.NoError(t, err)
+
+	f, err := pcap.OpenOffline("testdata/pcap/ipfix_cisco.pcap")
+	require.NoError(t, err)
+	defer f.Close()
+
+	var totalBytes, totalPackets int
+	limiter := rate.NewLimiter(rate.Limit(10000), 1)
+
+	packetSource := gopacket.NewPacketSource(f, f.LinkType())
+	for pkt := range packetSource.Packets() {
+		for !limiter.Allow() {
+		}
+
+		payloadData := pkt.TransportLayer().LayerPayload()
+
+		n, err := conn.Write(payloadData)
+		require.NoError(t, err)
+
+		totalBytes += n
+		totalPackets++
+	}
+
+	require.Zero(t, discardedEventsTotalVar.Get())
+
+	require.Eventually(t, func() bool {
+		return receivedEventTotalVar.Get() == uint64(totalPackets)
+	}, 10*time.Second, 200*time.Millisecond)
```

Comment:
why 10 secs? why copy pasted all over? could this be avoided?

Reply:
why not 10 secs? what is your beneficial counter argument here?

Reply:
Any fixed timing in the tests, is often a point of the test flakiness down the road or unnecessary long test run, which compounds overall tests run time

Reply:
if there is a really need for the delay, maybe make a constant that is easier to change if needed in one place

Reply:
added constants 🙂